server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchResizableBlockingQueue.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.common.util.concurrent;
+
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * A size based queue wrapping another blocking queue to provide (somewhat relaxed) capacity checks.
+ * Mainly makes sense to use with blocking queues that are unbounded to provide the ability to do
+ * capacity verification.
+ */
+public class OpenSearchResizableBlockingQueue<E> extends SizeBlockingQueue<E> {
```

Comment:
What are the reasons to introduce this class instead of using the existing `ResizableBlockingQueue` ?